lib/telephony/pinpoint/sms_sender.rb

```diff
@@ -45,7 +45,11 @@ def send(message:, to:, country_code:, otp: nil)
           )
           finish = Time.zone.now
           response = build_response(pinpoint_response, start: start, finish: finish)
-          return response if response.success?
+          if response.success? ||
+             response.error.is_a?(OptOutError) ||
+             response.error.is_a?(PermanentFailureError)
```

Comment:
For considering `PermanetnFailureError`, should we limit it to the cases like what we're matching with this logic?

https://github.com/18F/identity-idp/blob/12f214fdb50dc2af4b819dff7fc9e820ca90e106/lib/telephony/pinpoint/sms_sender.rb#L177-L183

On the other hand, if it is a permanent failure, regardless the specific, maybe we don't want failover anyways.